zulipterminal/model.py

```diff
@@ -563,9 +563,14 @@ def get_all_users(self) -> List[Dict[str, Any]]:
     @staticmethod
     def _stream_info_from_subscriptions(
             subscriptions: List[Dict[str, Any]]
-    ) -> Tuple[Dict[int, Any], Set[int], List[List[str]], List[List[str]]]:
-        stream_keys = ('name', 'stream_id', 'color',
-                       'invite_only', 'description')
+    ) -> Tuple[Dict[int, Any], Set[int], List[StreamData], List[StreamData]]:
+
+        def make_reduced_stream_data(stream: Dict[str, Any]) -> StreamData:
+            return StreamData({'name': stream['name'],
+                               'id': stream['stream_id'],
+                               'color': stream['color'],
+                               'invite_only': stream['invite_only'],
+                               'description': stream['description']})
```

Comment:
Something very odd here - I think it could be a mypy bug - I can't use dictionary comprehension or even a for loop to generate this dictionary being passed inside `StreamData`.

This causes it to fail
```python3
 sorted([StreamData({key: stream[key] for key in stream_keys})
                    for stream in subscriptions if not stream['pin_to_top']],
                   key=lambda s: s['name'].lower()),
```